sched/sched/sched_note.c

```diff
@@ -120,20 +120,21 @@ static inline void sched_note_flatten(FAR uint8_t *dst,
 {
   switch (len)
     {
-#ifdef CONFIG_HAVE_LONG_LONG
       case 8:
-        dst[7] = (uint8_t)((*(uint64_t *)src >> 56) & 0xff);
-        dst[6] = (uint8_t)((*(uint64_t *)src >> 48) & 0xff);
-        dst[5] = (uint8_t)((*(uint64_t *)src >> 40) & 0xff);
-        dst[4] = (uint8_t)((*(uint64_t *)src >> 32) & 0xff);
-#endif
       case 4:
-        dst[3] = (uint8_t)((*(uint32_t *)src >> 24) & 0xff);
-        dst[2] = (uint8_t)((*(uint32_t *)src >> 16) & 0xff);
       case 2:
-        dst[1] = (uint8_t)((*(uint16_t *)src >> 8) & 0xff);
       case 1:
-        dst[0] = (uint8_t)(*(uint8_t *)src & 0xff);
+#ifdef CONFIG_ENDIAN_BIG
```

Comment:
let's remove switch

Reply:
Switch partially ensures that we are handling standard types only.

Reply:
But it's just an internal function, we can trust caller.

Reply:
makes sense

Reply:
we can leave `DEBUGASSERT(len == 1 || len == 2 || len == 4 || len == 8);` optionally.